### PR TITLE
Recommendations v2

### PR DESCRIPTION
**Pull request by zainab-amir**
### Description

- Get launch ready for 100% users. 
- These recommendations will include both static and personalized recommendations. 
   - For static, we will be displaying Popular and Trending products.
   - For personalized, we will be quering Algolia
      - If Algolia is recommendations are skipped, only static are shown.

#### JIRA

[VAN-1597](https://2u-internal.atlassian.net/browse/VAN-1597)

#### How Has This Been Tested?

Locally and by adding new tests

#### Screenshots/sandbox (optional):

Screenshot 2023-08-21 at 12 23 25 PM

### CODE REVIEW

**Comment by zainab-amir** [comment]
We will not be showing static recommendations anymore. See https://2u-internal.atlassian.net/browse/VAN-1652 for more details.
|Screen|Updated Design|
|------|--------------|
|Large|Screenshot 2023-09-07 at 12 15 36 AM|
|Medium|Screenshot 2023-09-07 at 12 15 58 AM
|Small|Screenshot 2023-09-07 at 12 16 23 AM

File: src/recommendations/RecommendationsPage.jsx [hunk 1]
```diff
@@ -41,88 +44,72 @@ const RecommendationsPage = ({ countryCode }) => {
...
+                isLoading={isLoading}
+                personalizedRecommendations={algoliaRecommendations}
+              />
+            ) : (
+              <RecommendationsLargeLayout
+                userId={userId}
+                isLoading={isLoading}
+                personalizedRecommendations={algoliaRecommendations}
+              />
+            )}
```
Comment (syedsajjadkazmii) [inline src/recommendations/RecommendationsPage.jsx:88]: I don't think at this point with only personalized recommendations, we need small and large layouts. We can remove those layout and can have recommendations rendered here. Thoughts?

File: src/recommendations/data/hooks/useRecommendations.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import useAlgoliaRecommendations from './useAlgoliaRecommendations';
+
+/**
+ * This hooks returns Algolia and Static recommendations (Trending and Popular products).
+ * - Static recommendations are always returned.
+ * - Algolia recommendations are returned only if functional cookies are enabled and algolia returns recommendations.
+ * @param educationLevel
+ * @returns {{isLoading: boolean, trendingProducts: *[], popularProducts: *[], algoliaRecommendations: *[]}}
+ */
+export default function useRecommendations(educationLevel) {
```
Comment (syedsajjadkazmii) [inline src/recommendations/data/hooks/useRecommendations.jsx:12]: we dont need `useRecommendations` wrapper now, we can directly have country in the `useAlgoliaRecommendations` hook and we can use this hook directly in recommendations component.

File: src/recommendations/RecommendationsPage.jsx (hunk 1 above)
Comment (zainab-amir) [reply src/recommendations/RecommendationsPage.jsx:88]: I think this division is okay. The designs for small and large screen is different and we will still need to have two separate blocks of code to handle that. Its better that these are encapsulated

Comment (syedsajjadkazmii) [reply src/recommendations/RecommendationsPage.jsx:93]: How are the designs different? its just the cards that are wrapping to new line as width gets smaller. Do you want to discuss this on call?

Comment (zainab-amir) [reply src/recommendations/RecommendationsPage.jsx:93]: Sure

**Review by syedsajjadkazmii — APPROVED** [review]

**Comment by syedsajjadkazmii** [comment]
Please fix tests

**Pull request merged**